File: JAVA/08_Math1/BOJ_02_2292_벌집.java
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int cnt = 1, range = 2;
        if (n == 1) {
            System.out.println(1);
        } else {
            while (range <= n) {
                range += 6 * cnt;
                cnt++;
            }
            System.out.println(cnt);
        }
    }
}

File: C/08_Math1/BOJ_02_2292_벌집.c
#include <stdio.h>
main() {
	int n, cnt = 1, range = 2;
	scanf("%d", &n);
	if (n == 1) {
		printf("1");
	}
	else {
		while (range <= n) {
			range += 6 * cnt;
			cnt++;
		}
		printf("%d", cnt);
	}
}